### DISCUSSION TITLE
Remove feature received from vectorSource.getFeatureById.

### DISCUSSION THREAD

**Discussion by Kkalentiev** [openlayers/openlayers#15378] [Q&A] [answered]
After updating to OL `8.2.0`, I need to use `instanceOf` to remove feature received from vectorSource.getFeatureById.

```ts
const feature = this.vectorSource.getFeatureById(item.id); // return Feature<Geometry> | RenderFeature[]
if (feature instanceof Feature) this.vectorSource.removeFeature(feature); // require Feature<Geometry> by default
```

Is there any way to get feature by Id and delete it without using `instanceOf` as before?

**Comment by ahocevar** [comment, marked as answer]
I just created #15379 to improve this, by using conditional types based on the `FeatureClass` template. With that change, a `VectorSource` with `Feature` instances knows that `getFeatureById()` always returns a single `Feature`.

**Comment by Kkalentiev** [comment]
Thanks for the quick response!
